server/config.go

```diff
@@ -366,7 +368,7 @@ func (c *Config) genEmbedEtcdConfig() (*embed.Config, error) {
 	cfg.InitialCluster = c.InitialCluster
 	// Use unique cluster id as the etcd cluster token too.
 	cfg.InitialClusterToken = fmt.Sprintf("pd-%d", c.ClusterID)
-	cfg.ClusterState = embed.ClusterStateFlagNew
+	cfg.ClusterState = c.InitialClusterState
```

Comment:
I think we should remove c.InitialClusterState and use embed.ClusterStateFlagNew instead.